includes/CLI.php

```diff
@@ -61,26 +79,45 @@ public static function orders( $args, $assoc_args ) {
 		if ( ! empty( $assoc_args['status'] ) ) {
 			$status = $assoc_args['status'];
 			if ( ! wc_is_order_status( 'wc-' . $status ) ) {
-				WP_CLI::log( "The argument \"$status\" is not a valid order status." );
+				WP_CLI::error( "The argument \"$status\" is not a valid order status." );
 				return;
 			}
 		}
 
-		if ( $amount > 0 ) {
-			Generator\Order::disable_emails();
-			$progress = \WP_CLI\Utils\make_progress_bar( 'Generating orders', $amount );
-			for ( $i = 1; $i <= $amount; $i++ ) {
-				Generator\Order::generate( true, $assoc_args );
+		Generator\Order::disable_emails();
+
+		$progress = \WP_CLI\Utils\make_progress_bar( 'Generating orders', $amount );
+
+		add_action(
+			'smoothgenerator_order_generated',
+			function () use ( $progress ) {
 				$progress->tick();
 			}
-			$progress->finish();
+		);
+
+		$remaining_amount = $amount;
+		$generated        = 0;
+
+		while ( $remaining_amount > 0 ) {
+			$batch = $remaining_amount > Generator\Product::MAX_BATCH_SIZE ? Generator\Product::MAX_BATCH_SIZE : $remaining_amount;
```

Comment:
It currently amounts to the same thing, because both generators inherit this constant, but should this actually be:

```suggestion
			$batch = $remaining_amount > Generator\Order::MAX_BATCH_SIZE ? Generator\Order::MAX_BATCH_SIZE : $remaining_amount;
```

Reply:
Oops! Good catch on the copy pasta